server.go

```diff
@@ -1008,10 +1026,21 @@ func (s *server) Start() error {
 		return nil
 	}
 
+	// If one sub system fails to start, the following code ensures that the
+	// previous started ones are stopped. It also ensures a proper wallet
+	// shutdown which is important for releasing its resources (boltdb, etc...)
+	cleanup := cleaner{
+		s.cc.chainView.Stop,
```

Comment:
these services are not started w/in the context of the server, thus there cleanup should occur outside the context of the server's stop. they should be cleaned up after all server subsystems have exited

Reply:
@cfromknecht I agree, just keep in mind stopping these services was already handled in the `server.Stop` function before this PR.
Anyway I was thinking since they are created here: https://github.com/lightningnetwork/lnd/blob/master/chainregistry.go#L130 maybe good approach is to stop them as part of the `cleanup` function returned there.
Calling to `cleanup` already handled by `lnd.go` and then I can safely remove the logic that stops them in `server.Stop`
LMK what you think.

Reply:
@roeierez you are indeed correct, they are currently handled in the server even though they should probably be lifted out. I would support the approach you described and moving them to the lnd.go level cleanup assuming it doesn't introduce any weirdness :)

Reply:
@cfromknecht see my last commit. I explained the reasons in the commit message. 
LMK you feedback please.